test/integration/integration.cc

```diff
@@ -918,6 +918,59 @@ void BaseIntegrationTest::testNoHost() {
   EXPECT_TRUE(response.find("HTTP/1.1 400 Bad Request\r\n") == 0);
 }
 
+void BaseIntegrationTest::testAbsolutePath() {
+  Buffer::OwnedImpl buffer("GET http://www.redirect.com HTTP/1.1\r\nHost: host\r\n\r\n");
+  std::string response;
+  RawConnectionDriver connection(
+      lookupPort("http_forward"), buffer,
+      [&](Network::ClientConnection& client, const Buffer::Instance& data) -> void {
+        response.append(TestUtility::bufferToString(data));
+        client.close(Network::ConnectionCloseType::NoFlush);
+      },
+      version_);
+
+  connection.run();
+  EXPECT_FALSE(response.find("HTTP/1.1 404 Not Found\r\n") == 0);
+}
+
+void BaseIntegrationTest::testAllowAbsoluteSameRelative() {
+  // Ensure that relative urls behave the same with allow_absolute_url enabled and without
+  testEquivalent("GET /foo/bar HTTP/1.1\r\nHost: host\r\n\r\n");
+}
+
+void BaseIntegrationTest::testConnect() {
+  // Ensure that connect behaves the same with allow_absolute_url enabled and without
+  testEquivalent("CONNECT www.somewhere.com:80 HTTP/1.1\r\nHost: host\r\n\r\n");
+}
+
+void BaseIntegrationTest::testEquivalent(std::string request) {
```

Comment:
From a quick glance I'm not sure what "testEquivalent" does. Can you change name to something more descriptive or add comment?